census_extractomatic/api.py

```diff
@@ -2009,6 +2021,303 @@ def table_geo_comparison_rowcount(table_id):
 
     return resp
 
+## COMBINED LOOKUPS ##
+
+@app.route("/2.1/full-text/search")
+@qwarg_validate({
+    'q':   {'valid': NonemptyString()},
+    'type': {'valid': OneOf(allowed_searches), 'default': allowed_searches[3]},
+})
+@crossdomain(origin='*')
+def full_text_search():
+
+    def do_search(db, q, object_type):
+        """ Search for objects (profiles, tables, topics) matching query q.
+
+        Return a list, because it's easier to work with than a SQLAlchemy 
+        ResultProxy object (notably, the latter does not support indexing).
+        """
+
+        q_profiles = """SELECT text1 AS display_name, 
+                               text2 AS sumlevel,
+                               text3 AS sumlevel_name,
+                               text4 AS full_geoid,
+                               text5 AS population, 
+                               text6 AS priority,
+                               ts_rank(document, to_tsquery('simple', '{0}')) AS relevance,
+                               type
+                        FROM search_metadata
+                        WHERE document @@ to_tsquery('simple', '{0}')
+                        AND type = 'profile'
+                        ORDER BY CAST(text6 as INT) ASC, 
+                                 CAST(text5 as INT) DESC, 
+                                 relevance DESC;""".format(q)
+
+        q_tables = """SELECT text1 AS tabulation_code, 
+                             text2 AS table_title,
+                             text3 AS topics,
+                             text4 AS simple_table_title,
+                             text5 AS tables,
+                             ts_rank(document, to_tsquery('{0}'), 2|8|32) AS relevance,
+                             type
+                      FROM search_metadata
+                      WHERE document @@ to_tsquery('{0}')
+                      AND type = 'table'
+                      ORDER BY relevance DESC;""".format(q)
+
+        q_topics = """SELECT text1 as topic_name,
+                             text3 as url,
+                             ts_rank(document, to_tsquery('{0}')) AS relevance,
+                             type
+                      FROM search_metadata
+                      WHERE document @@ to_tsquery('{0}')
+                      AND type = 'topic'
+                      ORDER BY relevance DESC;""".format(q)
+
+        q_map = {'profile': q_profiles, 'table': q_tables, 'topic': q_topics}
```

Comment:
Let's just use an if-elif chain in this method so that we don't spend time creating 3 string objects and then only use one of them.